lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -11,84 +11,1938 @@
+        arrayIndexOf(array, el, from) { return callInstanceFunc(ArrayInstanceIndexOf, array, el, from); },
+        map(array, func) { return callInstanceFunc(platform.builtInJavascriptArrayEntryMap, array, func); },
+        reduce(array, func, init) { return callInstanceFunc(platform.builtInJavascriptArrayEntryReduce, array, func, init); },
+        slice(array, start, end) { return callInstanceFunc(platform.builtInJavascriptArrayEntrySlice, array, start, end); },
+        concat(array, ...els) { return callInstanceFunc(platform.builtInJavascriptArrayEntryConcat, array, ...els); },
+        filter(array, func) { return callInstanceFunc(platform.builtInJavascriptArrayEntryFilter, array, func); },
+        unique(array) { return _.filter(array, (v, i) => _.arrayIndexOf(array, v) === i); },
+
+        keys: platform.builtInJavascriptObjectEntryKeys,
+        hasOwnProperty(o, prop) { return callInstanceFunc(platform.builtInJavascriptObjectEntryHasOwnProperty, o, prop); },
+        defineProperty: platform.builtInJavascriptObjectEntryDefineProperty,
+        isExtensible: platform.builtInJavascriptObjectEntryIsExtensible,
+        create(proto = null) { return platform.builtInJavascriptObjectCreate(proto); },
+        setPrototypeOf(target, proto = null) { return platform.builtInSetPrototype(target, proto); },
+
+        abs: platform.builtInMathAbs,
+        floor: platform.builtInMathFloor,
+        max: platform.builtInMathMax,
+        pow: platform.builtInMathPow,
+
+        isFinite: platform.builtInGlobalObjectEntryIsFinite,
+        isNaN: platform.builtInGlobalObjectEntryIsNaN,
+
+        getDate(date) { return callInstanceFunc(platform.builtInJavascriptDateEntryGetDate, date); },
+
+        bind(func, that) { return callInstanceFunc(platform.builtInJavascriptFunctionEntryBind, func, that); },
+        apply(func, that, args) { return callInstanceFunc(platform.builtInJavascriptFunctionEntryApply, func, that, args); },
+    };
+
+    const raise = {
+        rangeError() { return arguments.length === 3 ? platform.raiseOptionValueOutOfRange_3(...arguments) : platform.raiseOptionValueOutOfRange(); },
+        assert(test, err) { return test ? undefined : platform.raiseAssert(err || new Error("Assert failed")); }
+    };
+
+    // When this file was originally written, it assumed Windows Globalization semantics.
+    // Throughout the transition to ICU, we tried to share as much code as possible between WinGlob and ICU.
+    // However, because ICU has different semantics and our ICU-based implementation tries to match a newer
+    // version of the Intl spec, we have decided that the code sharing was causing more harm than good.
+    // Thus, while we support both ICU and WinGlob, we have decided to duplicate a substantial amount of code.
+    // The indentation of the below if block is intentionally incorrect so as to minimize diff.
+    if (isPlatformUsingICU) {
```

Comment:
Since this is easy to miss, when reviewing this file, note that most of the "added" code below is simply duplicated between isPlatformUsingICU and isPlatformUsingWinGlob.